### DISCUSSION TITLE
time-resolved simulation questions

### DISCUSSION THREAD

**Discussion by codycrounds** [ankrh/MCmatlab#12] [General]
Hi Anders, 

I hope you are doing well, and thank you for all the advice you have provided in the past with your MCMatlab software. I’m reaching out again about another snag I’ve hit that I’m hoping you could help sort out. We are hoping to do some time resolved measurements, and thus are working off of your time resolved example. However, I’ve noticed two things: 
1) the time resolved fluorescence mode seems to have a bug in it, where the model.MC.LC.image for this version of the simulations is only 2D and doesn’t yield the third time dimension. To manage this, I’ve been using the generic MC mode where I use a point source as the light source and this yields the time-resolved measurements in the model.MC.LC.image variable. Are you aware of how we might be able to correct this for the fluorescence example? 

2) I’ve noticed that the simulation assumes that the initial pulse of light is infinitely small, and thus doesn’t account for temporal spread functions as seen in lasers, and as such cannot take into consideration a fluorophore’s lifetime. Are you aware of a way to manage this in the simulations, in a way that we can extend the pulse to reflect the lifetime of a fluorophore? 

Thank you! 

Cody Rounds

**Comment by ankrh** [comment]
Hi Cody!

Sorry that I didn't get around to answering this question. It slipped my mind until now.

The lack of time-resolution for fluorescence is an intentional design choice (for now), although MCmatlab probably should have thrown a warning explaining this.

I think your simulation corresponds to calculating a sort of "escape time" of emitted fluorescence from each voxel in the cuboid. But that's only half the truth. You'd need to convolve that with the distribution of "arrival times" of the excitation light in each voxel and then also convolve that with the fluorescence emission exponential function. Additionally, you could convolve this result with the input pulse shape.

The "arrival time" (a 4D normalized absorption array x,y,z,t) is something that MCmatlab can't calculate yet, although I have thought of implementing it. It would be very memory demanding.
